File: Models/DsngayDocSoBillingChiTiet.cs
using System;
using System.Collections.Generic;

namespace WebAPI_NRW.Models.Database;

public partial class DsngayDocSoBillingChiTiet
{
    public int Id { get; set; }

    public int? MaNgayDocSo { get; set; }

    public int Nam { get; set; }

    public int Ky { get; set; }

    public int Dot { get; set; }

    public int SoNgayDocSoDot { get; set; }

    public string? GhiChu { get; set; }

    public DateTime NgayTao { get; set; }

    public DateTime? NgayCapNhat { get; set; }

    public int NguoiTao { get; set; }

    public int? NguoiCapNhat { get; set; }

    public virtual DsngayDocSoBilling? MaNgayDocSoNavigation { get; set; }

    public virtual NguoiDung? NguoiCapNhatNavigation { get; set; }

    public virtual NguoiDung NguoiTaoNavigation { get; set; } = null!;
}

File: src/types/he-thong-billing/ds-ngay-doc-so-billing-chi-tiet.ts
/* ========== REQUEST (CLIENT --> API) ========== */
// Thêm dữ liệu
export interface AddDsNgayDocSoBillingChiTietRequest {
  maNgayDocSo?: string;  // string, giống BE
  nam: number;
  ky: number;
  dot: number;
  soNgayDocSoDot: number;
  ghiChu?: string;
  ngayTao: string;
  nguoiTao: string;      // string, giống BE
}

// Cập nhật dữ liệu
export interface UpdateDsNgayDocSoBillingChiTietRequest {
  maNgayDocSo?: string;  // string, giống BE
  nam: number;
  ky: number;
  dot: number;
  soNgayDocSoDot: number;
  ghiChu?: string;
  ngayCapNhat?: string;
  nguoiCapNhat?: string; // string, giống BE
}

/* ========== RESPONSE (API --> CLIENT) ========== */
export interface DsNgayDocSoBillingChiTietResponse {
  id: number;
  maNgayDocSo?: string;
  nam: number;
  ky: number;
  dot: number;
  soNgayDocSoDot: number;
  ghiChu?: string;
  ngayTao: string;
  ngayCapNhat?: string;
  nguoiTao?: string;
  nguoiCapNhat?: string;
}
